### PR TITLE
Expand cli documentation relating to the no-input flag (#1543)

### PR DESCRIPTION
**Pull request by jeremyswerdlow**
Addresses the concerns in #1543 that not all side effects are clearly documented

### CODE REVIEW

**Comment by glumia** [comment]
Hey @jeremyswerdlow thanks for taking time for this, and sorry for the delay on the review!

While we're at it, I'd like to be able to fix other problems with flag's documentation within the codebase.  

I tracked down all the places where `no_input` appears, and this is what I noticed:

- cookiecutter/main.py::cookiecutter
- cookiecutter/prompt.py::prompt_for_config
- cookiecutter/prompt.py::prompt_choice_for_config
- cookiecutter/repository.py::determine_repo_dir

In all those places the docstring just says *"Prompt the user at command line for
manual configuration?"*. But as a user of this function, I ask myself: what
happens if the end-user isn't prompted for manual configuration? (I
should be able to understand it by looking only at the docstring) 


- cookiecutter/zipfile.py::unzip

*"Suppress any prompts"* doesn't say anything about how the function
behaves in this case. Take a look at cookiecutter/utils.py::prompt_and_delete, 
this is an example of good documentation: if I read the docstring I
don't have to guess what happens.


- cookiecutter/vcs.py::clone

Same as above, how does the function behave if the flag is `True`? (there's no mention
about the cache being refreshed).


- docs/advanced/suppressing_prompts.rst:6 

We should document also here the fact that `no_input` causes a refresh of the cached resources. 
I would also add a comment on line 6 to say that `no_input` corresponds to `--no-input` flag on the CLI. 


Would you mind taking care of them?

**Comment by glumia** [comment]
Hey @jeremyswerdlow are you still interested on working on this? 

I don't want to put pressure, it's just to know if I should wait or add those changes by myself and merge the pr.

File: cookiecutter/prompt.py [hunk 1]
```diff
@@ -168,10 +168,9 @@ def render_variable(env, raw, cookiecutter_dict):
 def prompt_choice_for_config(cookiecutter_dict, env, key, options, no_input):
     """Prompt user with a set of options to choose from.
 
-    Each of the possible choices is rendered beforehand.
+    :no_input: Do not prompt for user input and return the first available option.
```
Comment (insspb) [inline cookiecutter/prompt.py:171]:
```suggestion
    :param no_input: Do not prompt for user input and return the first available option.
```

**Review by insspb — APPROVED** [review]

**Comment by insspb** [comment]
@jeremyswerdlow @glumia Thank you for contribution, will be in next release.

**Pull request merged**